rwatch/ui/animation/property_animation.c

```diff
@@ -0,0 +1,172 @@
+/* property_animation.c
+ * a convenience animation for 
+ * libRebbleOS
+ *
+ * Author: Carson Katri <me@carsonkatri.com>
+ */
+
+#include "librebble.h"
+#include "appmanager.h"
+#include "FreeRTOS.h"
+#include "property_animation.h"
+#include "animation.h"
+
+void property_animation_update_grect(PropertyAnimation * property_animation, const uint32_t distance_normalized)
+{
+    if (property_animation->impl.accessors.getter.grect != NULL && property_animation->impl.accessors.setter.grect != NULL)
```

Comment:
(informational; I'll merge anyway)

Better to invert the sense of this:

```
  if (blah == NULL || blah2 == NULL)
    return;
  everything else...
```

Or perhaps:

```
  if (!blah || !blah2)
    return;
  everything else...
```